File: api/estimate.py
from http.server import BaseHTTPRequestHandler
import json
import urllib.request
import urllib.parse

class handler(BaseHTTPRequestHandler):
    def do_POST(self):
        try:
            content_length = int(self.headers.get('Content-Length', 0))
            post_data = self.rfile.read(content_length)
            data = json.loads(post_data.decode('utf-8'))
            
            last_msg = data.get('prompt', 'Hello')
            system_prompt = data.get('system', '')
            
            encoded_prompt = urllib.parse.quote(last_msg)
            encoded_system = urllib.parse.quote(system_prompt)
            url = f"https://text.pollinations.ai/{encoded_prompt}?system={encoded_system}"
            
            req = urllib.request.Request(
                url,
                headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) TASKAS/1.0'}
            )
            with urllib.request.urlopen(req, timeout=12) as response:
                reply_text = response.read().decode('utf-8')
            
            self.send_response(200)
            self.send_header('Content-Type', 'application/json')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(json.dumps({'response': reply_text}).encode('utf-8'))
        except Exception as e:
            self.send_response(500)
            self.send_header('Content-Type', 'application/json')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(json.dumps({'error': str(e)}).encode('utf-8'))

File: functions/api/estimate.js
export async function onRequestOptions(context) {
  return new Response(null, {
    status: 204,
    headers: {
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Methods': 'POST, OPTIONS',
      'Access-Control-Allow-Headers': 'Content-Type',
      'Access-Control-Max-Age': '86400',
    }
  });
}

export async function onRequestPost(context) {
  const corsHeaders = {
    'Access-Control-Allow-Origin': '*',
    'Content-Type': 'application/json',
  };

  try {
    const { request } = context;
    const data = await request.json();
    
    const lastMsg = data.prompt || 'Hello';
    const systemPrompt = data.system || '';
    
    const encodedPrompt = encodeURIComponent(lastMsg);
    const encodedSystem = encodeURIComponent(systemPrompt);
    const url = `https://text.pollinations.ai/${encodedPrompt}?system=${encodedSystem}`;
    
    const response = await fetch(url, {
      method: 'GET',
      headers: {
        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) TASKAS/1.0'
      }
    });
    
    if (!response.ok) {
      throw new Error(`Pollinations API returned status: ${response.status}`);
    }
    
    const replyText = await response.text();
    
    return new Response(JSON.stringify({ response: replyText }), {
      status: 200,
      headers: corsHeaders
    });
  } catch (e) {
    return new Response(JSON.stringify({ error: e.message }), {
      status: 500,
      headers: corsHeaders
    });
  }
}
